Source/WebCore/CMakeLists.txt

```diff
@@ -2296,13 +2296,20 @@ set(WebCoreTestSupport_INTERFACE_INCLUDE_DIRECTORIES
 )
 
 WEBKIT_FRAMEWORK_DECLARE(WebCore)
-WEBKIT_FRAMEWORK_DECLARE(WebCoreTestSupport)
+WEBKIT_LIBRARY_DECLARE(WebCoreTestSupport)
 WEBKIT_INCLUDE_CONFIG_FILES_IF_EXISTS()
 
+if (WebCore_LIBRARY_TYPE STREQUAL SHARED)
+    list(APPEND WebCoreTestSupport_PRIVATE_LIBRARIES WebCore)
+else ()
+    list(APPEND WebCoreTestSupport_LIBRARIES ${WebCore_LIBRARIES})
+    list(APPEND WebCoreTestSupport_PRIVATE_LIBRARIES ${WebCore_PRIVATE_LIBRARIES})
+    list(APPEND WebCoreTestSupport_DEPENDENCIES WebCore)
```

Comment:
Two issues here. `WebCoreTestSupport` needs includes from other libraries like ICU, harfbuzz, etc, to build properly. So thats why the `_LIBRARIES` and `_PRIVATE_LIBRARIES` are used. If we add `WebCore` to `_PRIVATE_LIBRARIES` then it ends up with a library or OBJECT files that it would then need to link so `_DEPENDENCIES` is used instead so its required for the build.

Also as a note these were moved up because if they come after `WEBKIT_FRAMEWORK(WebCore)` those variables are modified with the results of propagating the `_FRAMEWORKS` variable.

Reply:
Hmm, it's too difficult to understand to me.
`WebCore` is declared as a FRAMEWORK.
Why can't we write as:
> list(APPEND WebCoreTestSupport_FRAMEWORK WebCore)
Why can't your FRAMEWORK system automatically solve those problems?